flutter-provider/client/lib/models/auth.dart

```diff
@@ -0,0 +1,35 @@
+import 'package:flutter/foundation.dart';
+import 'package:provider_experiment/client.dart';
+import 'package:shared_preferences/shared_preferences.dart';
+
+class AuthModel extends ChangeNotifier {
+  final SharedPreferences _prefs;
+
+  AuthModel(this._prefs);
+
+  bool get isAuthenticated {
+    return _prefs.getBool('isAuthenticated') ?? false;
+  }
+
+  set isAuthenticated(bool val) {
+    _prefs.setBool('isAuthenticated', val);
+    notifyListeners();
+  }
+
+  logout() {
+    isAuthenticated = false;
+  }
+
+  Future<void> login(String user, String pwd) async {
+    final res = await Client().init().post("/login", data: {
```

Comment:
Maybe not for this cycle, but would make sense to use a service locator pattern with get_it for example. I'd love to see it instead of calling the constructor here. Of course, that would result in better unit tests, something that you are not caring about here.

Reply:
The idea was to create a new dio instance, we can't reuse it really

Reply:
@wesleyfuchter Have a look, I've added `get_it` in my last commit